Makefile

```diff
@@ -7,63 +7,112 @@ all: build
+# to get around that, we "escape" those characters by converting all : to --- and all / to ___ , so that we can use them
+# in the target, we then unescape them back
+escapefs = $(subst :,---,$(subst /,___,$(1)))
+unescapefs = $(subst ---,:,$(subst ___,/,$(1)))
+
+# Targets used when cross building.
+.PHONY: native register
+native:
+ifneq ($(BUILDARCH),$(ARCH))
+	@echo "Target $(MAKECMDGOALS)" is not supported when cross building! && false
 endif
 
-ifeq ($(ARCH),s390x)
-	ARCHTAG:=-s390x
+# Enable binfmt adding support for miscellaneous binary formats.
+# This is only needed when running non-native binaries.
+register:
+ifneq ($(BUILDARCH),$(ARCH))
+	docker run --rm --privileged multiarch/qemu-user-static:register || true
 endif
+
+
+# list of arches *not* to build when doing *-all
+#    until s390x works correctly
+EXCLUDEARCH ?= s390x
+VALIDARCHES = $(filter-out $(EXCLUDEARCH),$(ARCHES))
+
 ###############################################################################
+CONTAINER_BASENAME ?= node
+CONTAINER_NAME ?= calico/$(CONTAINER_BASENAME)
+
 GO_BUILD_VER?=v0.17
 CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)
 
 # Version of this repository as reported by git.
 CALICO_GIT_VER := $(shell git describe --tags --dirty --always)
 
 # Versions and location of dependencies used in the build.
-BIRD_VER?=v0.3.2
-BIRD_URL?=https://github.com/projectcalico/calico-bird/releases/download/$(BIRD_VER)/bird
-BIRD6_URL?=https://github.com/projectcalico/calico-bird/releases/download/$(BIRD_VER)/bird6
-BIRDCL_URL?=https://github.com/projectcalico/calico-bird/releases/download/$(BIRD_VER)/birdcl
+BIRD_VER?=v0.3.2-13-g17d14e60
+BIRD_IMAGE ?= calico/bird:$(BIRD_VER)-$(ARCH)
 
 # Versions and locations of dependencies used in tests.
 CALICOCTL_VER?=master
 CNI_VER?=master
 RR_VER?=master
 TEST_CONTAINER_NAME_VER?=latest
-CTL_CONTAINER_NAME?=calico/ctl$(ARCHTAG):$(CALICOCTL_VER)
-RR_CONTAINER_NAME?=calico/routereflector$(ARCHTAG)
-TEST_CONTAINER_NAME?=calico/test$(ARCHTAG):$(TEST_CONTAINER_NAME_VER)
+CTL_CONTAINER_NAME?=calico/ctl:$(CALICOCTL_VER)-$(ARCH)
+RR_CONTAINER_NAME ?= calico/routereflector:$(RR_VER)
+TEST_CONTAINER_NAME?=calico/test:$(TEST_CONTAINER_NAME_VER)-$(ARCH)
 ETCD_VERSION?=v3.3.7
-ETCD_IMAGE?=quay.io/coreos/etcd:$(ETCD_VERSION)$(ARCHTAG)
+# If building on amd64 omit the arch in the container name.  Fixme!
+ETCD_IMAGE?=quay.io/coreos/etcd:$(ETCD_VERSION)
+ifneq ($(BUILDARCH),amd64)
+        ETCD_IMAGE=$(ETCD_IMAGE)-$(ARCH)
+endif
+
 K8S_VERSION?=v1.10.4
 HYPERKUBE_IMAGE?=gcr.io/google_containers/hyperkube-$(ARCH):$(K8S_VERSION)
 TEST_CONTAINER_FILES=$(shell find tests/ -type f ! -name '*.created')
 
+# Variables controlling the calico/node image
+NODE_CONTAINER_CREATED=.calico_node.created-$(ARCH)
+NODE_CONTAINER_BIN_DIR=./dist/bin/
+# files in the container that we create locally
+NODE_BINARIES_FILES=calico-node-$(ARCH)
+NODE_CONTAINER_BINARIES = $(addprefix $(NODE_CONTAINER_BIN_DIR),$(NODE_BINARIES_FILES))
```

Comment:
And this is only ever one file, so shouldn't be plural. I think You can just get rid of NODE_BINARIES_FILES and just put the value inline here